### PR TITLE
test:Validation and Logic Tests for ERPNext Available Serial No Report

### PR DESCRIPTION
**Pull request by hrishikesh8848**
### Title: 
Add Test Cases for Available Serial No Report

### Test Summary
This test suite validates the logic and correctness of the Available Serial No report in ERPNext. It ensures accurate reporting of available serial numbers based on filters and stock ledger entries, including edge case handling and data accuracy across item and warehouse combinations.

### Scenarios Covered

Input Validations
T_ASN_001 – Report with Valid Item and Warehouse
Checks that the report returns the correct serial numbers when both item and warehouse filters are applied.

T_ASN_002 – Report with Only Item Filter
Validates that the report returns available serial numbers across all warehouses when only an item is selected.

T_ASN_003 – Report with Only Warehouse Filter
Ensures correct serial number results across all items for the specified warehouse.

 Edge Cases
T_ASN_004 – Serial Numbers with No Stock Ledger Entries
Ensures such serial numbers are excluded from the report.

T_ASN_005 – Multiple Serial No Records for a Single Item
Confirms that all valid serial numbers are included and grouped appropriately.

T_ASN_006 – Serial No Filter on Empty Stock
Ensures the report handles empty results gracefully when no stock exists.

T_ASN_007 – Serial Nos with Zero or Negative Balance Qty
Verifies these serials are excluded from the report.

### Technology

Test Framework: unittest

Base Class: unittest.TestCase

Module Under Test:
erpnext.stock.report.available_serial_no.available_serial_no

Functions Tested:
execute, update_stock_ledger_entry

### Linked Feature/Bug
None

### CODE REVIEW

File: erpnext/stock/report/available_serial_no/test_available_serial_no.py [hunk 1]
```diff
@@ -34,12 +46,298 @@ def test_available_serial_no(self):
...
+	def test_invalid_item_code_T_ASN_008(self):
+		from erpnext.stock.report.available_serial_no import available_serial_no
+
+		columns, data = [], []
+		try:
+			columns, data = available_serial_no.execute(filters={"item_code": "Non-Existent Item"})
+		except Exception:
+			pass
+
+		self.assertEqual(len(data), 0)
```
Comment (aasif-patel) [inline erpnext/stock/report/available_serial_no/test_available_serial_no.py:201]: I think we can use assertRaises here to check the validation

**Review by aasif-patel — CHANGES_REQUESTED** [review]

**Comment by aasif-patel** [comment]
@hrishikesh8848 PR not following pre-commit rules hence closing
